validator/sawtooth_validator/state/client_handlers.py

```diff
@@ -382,6 +385,103 @@ def id_by_index(index, resources):
             return resources[index].address
 
 
+class _Sorter(object):
+    """A static class containing a method to sort lists of resources based on
+    SortControls sent with the request.
+    """
+    @classmethod
+    def sort_resources(cls, request, resources, fail_enum, header_proto=None):
+        """Sorts a list of resources based on a list of sort controls
+
+        Args:
+            request (object): The parsed protobuf request object
+            resources (list of objects): The resources to be sorted
+            fail_enum (int, enum): The enum status to raise with invalid keys
+            header_proto(class): Class to decode a resources header
+
+        Returns:
+            list: The sorted list of resources
+        """
+        if not request.sorting:
+            return resources
+
+        value_handlers = [cls._ValueHandler(controls, fail_enum, header_proto)
+                          for controls in request.sorting]
+
+        def sorter(resource_a, resource_b, value_handlers=value_handlers):
+            if not value_handlers:
+                return 0
+            handler = value_handlers[0]
+
+            value_a, value_b = handler.get_sort_values(resource_a, resource_b)
+
+            if value_a < value_b:
+                return handler.xform_result(-1)
+            if value_a > value_b:
+                return handler.xform_result(1)
+            return sorter(resource_a, resource_b, value_handlers[1:])
```

Comment:
SortControls are unhashable. The basic idea should still work, but I won't be able to use `set`